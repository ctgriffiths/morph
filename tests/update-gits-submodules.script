#!/bin/sh
#
# Test that 'morph update-gits' caches all the submodules needed by
# a chunk. 
#
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

set -eu

# create a repo to use as a submodule
submodule="$DATADIR/submodule-repo"
"$SRCDIR/scripts/cmd-filter" git init --quiet "$submodule"
"$SRCDIR/scripts/run-git-in" "$submodule" commit --quiet --allow-empty \
                                          --allow-empty-message -m ""

# create an empty commit in a chunk repository
chunkrepo="$DATADIR/chunk-repo"
"$SRCDIR/scripts/run-git-in" "$chunkrepo" checkout --quiet farrokh
"$SRCDIR/scripts/run-git-in" "$chunkrepo" submodule --quiet add "$submodule" \
    > /dev/null
"$SRCDIR/scripts/run-git-in" "$chunkrepo" commit --quiet -m "add submodule"

"$SRCDIR/scripts/test-morph" update-gits \
    test:morphs-repo master hello-stratum.morph

test -d "$DATADIR/cache/gits/"*chunk?repo* && echo chunk-repo cached
test -d "$DATADIR/cache/gits/"*morphs?repo* && echo morphs-repo cached
test -d "$DATADIR/cache/gits/"*submodule?repo* && echo submodule-repo cached
