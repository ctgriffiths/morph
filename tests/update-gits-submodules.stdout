chunk-repo cached
morphs-repo cached
submodule-repo cached
